File: src/main/scala/com/github/mauricio/postgresql/parsers/ReturningMessageParser.scala
package com.github.mauricio.postgresql.parsers

import org.jboss.netty.buffer.ChannelBuffer
import com.github.mauricio.postgresql.messages.backend.Message

/**
 * User: Maurício Linhares
 * Date: 3/12/12
 * Time: 11:36 PM
 */

class ReturningMessageParser( val message : Message ) extends MessageParser {
  def parseMessage(buffer: ChannelBuffer): Message = {
    this.message
  }
}

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/parsers/ReturningMessageParser.kt
package com.github.mauricio.async.db.postgresql.parsers

import com.github.mauricio.async.db.postgresql.messages.backend.BindComplete
import com.github.mauricio.async.db.postgresql.messages.backend.CloseComplete
import com.github.mauricio.async.db.postgresql.messages.backend.EmptyQueryString
import com.github.mauricio.async.db.postgresql.messages.backend.NoData
import com.github.mauricio.async.db.postgresql.messages.backend.ParseComplete
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage
import io.netty.buffer.ByteBuf

enum class ReturningMessageParser(val message: ServerMessage) : MessageParser {

  BindCompleteMessageParser(BindComplete),
  CloseCompleteMessageParser(CloseComplete),
  EmptyQueryStringMessageParser(EmptyQueryString),
  NoDataMessageParser(NoData),
  ParseCompleteMessageParser(ParseComplete);

  override fun parseMessage(buffer: ByteBuf): ServerMessage = this.message

}
